autotest/gcore/tiff_read_subifds.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env pytest
+# -*- coding: utf-8 -*-
+###############################################################################
+# $Id$
+#
+# Project:  GDAL/OGR Test Suite
+# Purpose:  Test basic read support for tiff files using overviews in subifds
+# Author:   Thomas Bonfort <thomas.bonfort@airbus.com>
+#
+###############################################################################
+# Copyright (c) 2019,  Thomas Bonfort <thomas.bonfort@airbus.com>
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Library General Public
+# License as published by the Free Software Foundation; either
+# version 2 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Library General Public License for more details.
+#
+# You should have received a copy of the GNU Library General Public
+# License along with this library; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+# Boston, MA 02111-1307, USA.
+###############################################################################
+
+import os
+import sys
+import shutil
+import struct
+
+import pytest
+
+
+import gdaltest
+from osgeo import gdal, osr
+
+
+###############################################################################
+# Test absolute/offset && index directory access
+
+
+def test_tiff_read_off():
+
+    ds = gdal.Open('data/tiff_with_subifds.tif')
+    assert ds.GetRasterBand(1).Checksum() == 35731
+
+    data = ds.GetRasterBand(1).ReadRaster(buf_xsize=1,buf_ysize=1, xsize=1, ysize=1,buf_type=gdal.GDT_Int16)
+    assert data[0][0] == "\xdc" #212
```

Comment:
data[0][0] is a bit of weird syntax, data being a flat byte buffer.
`assert struct.unpack('H', data)[0] == 212` would be better.